### PR TITLE
Fix ZFP cuda build.

### PR DESCRIPTION
**Pull request by corbett5**
Adding `+cuda` wouldn't actually build ZFP with CUDA.

@lindstro @GarrettDMorrison

### CODE REVIEW

**Comment by lindstro** [comment]
Good catch.  Not sure how we missed this.  The same applies to `+openmp`.  I will submit a separate PR to fix that.

File: var/spack/repos/builtin/packages/zfp/package.py [hunk 1]
```diff
@@ -99,7 +99,15 @@ def cmake_args(self):
             self.define_from_variant('ZFP_WITH_ALIGNED_ALLOC', 'aligned'),
             self.define_from_variant('ZFP_WITH_CACHE_TWOWAY', 'twoway'),
             self.define_from_variant('ZFP_WITH_CACHE_FAST_HASH', 'fasthash'),
-            self.define_from_variant('ZFP_WITH_CACHE_PROFILE', 'profile')
+            self.define_from_variant('ZFP_WITH_CACHE_PROFILE', 'profile'),
+            self.define_from_variant('ZFP_WITH_CUDA', 'cuda')
         ]
 
+        if '+cuda' in spec:
+            args.append('-DCUDA_BIN_DIR={}'.format(spec['cuda'].prefix.bin))
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/zfp/package.py]:
```suggestion
            args.append('-DCUDA_BIN_DIR={0}'.format(spec['cuda'].prefix.bin))
```

File: var/spack/repos/builtin/packages/zfp/package.py [hunk 2]
```diff
@@ -99,7 +99,15 @@ def cmake_args(self):
...
+
+            if not spec.satisfies('cuda_arch=none'):
+                cuda_arch = spec.variants['cuda_arch'].value
+                args.append('-DCMAKE_CUDA_FLAGS=-arch sm_{}'.format(cuda_arch[0]))
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/zfp/package.py]:
```suggestion
                args.append('-DCMAKE_CUDA_FLAGS=-arch sm_{0}'.format(cuda_arch[0]))
```

**Review by tldahlgren — CHANGES_REQUESTED** [review]

**Comment by tldahlgren** [comment]
Please fix the style check failure. I made a couple of suggestions that should help.

File: var/spack/repos/builtin/packages/zfp/package.py [hunk 3]
```diff
@@ -99,7 +99,15 @@ def cmake_args(self):
             self.define_from_variant('ZFP_WITH_ALIGNED_ALLOC', 'aligned'),
             self.define_from_variant('ZFP_WITH_CACHE_TWOWAY', 'twoway'),
             self.define_from_variant('ZFP_WITH_CACHE_FAST_HASH', 'fasthash'),
-            self.define_from_variant('ZFP_WITH_CACHE_PROFILE', 'profile')
+            self.define_from_variant('ZFP_WITH_CACHE_PROFILE', 'profile'),
+            self.define_from_variant('ZFP_WITH_CUDA', 'cuda')
```
Comment (lindstro) [inline var/spack/repos/builtin/packages/zfp/package.py]: Can we move the cuda variant up to just under the fortran variant?  I'd like to keep the advanced variants separate at the bottom.

File: var/spack/repos/builtin/packages/zfp/package.py (hunk 2 above)
Comment (corbett5) [reply var/spack/repos/builtin/packages/zfp/package.py]: Done.

File: var/spack/repos/builtin/packages/zfp/package.py (hunk 3 above)
Comment (corbett5) [reply var/spack/repos/builtin/packages/zfp/package.py]: Done.

**Review by lindstro — APPROVED** [review]

**Review by tldahlgren — APPROVED** [review]
LGTM.  Approval contingent on all required checks passing.

**Comment by corbett5** [comment]
All the checks passed, is there another reason it didn't get merged?

**Pull request merged**